### PR TITLE
feat : SQL expressions for URI in dimensions

### PR DESCRIPTION
**Pull request by k-anshul**
https://github.com/rilldata/rill/issues/5282

Allows caller to get URI for a dimension using a measure compute:
```
 {
      "name": "app_site_domain_uri",
      "uri": {
          "dimension": "app_site_domain"
      }
}
```
check `runtime/queries/metricsview_aggregation_test.go -> testMetricsViewsAggregationURI` for detailed request.

### CODE REVIEW

**Comment by mindspank** [comment]
@nishantmonu51 we'd need some tiny frontend changes to make use of the new backend property as well. Are we tracking that anywhere?

**Comment by k-anshul** [comment]
Hey @mindspank 

Just an FYI that we will change this approach. This has certain limitations.

File: proto/rill/runtime/v1/queries.proto [hunk 1]
```diff
@@ -404,6 +404,7 @@ message MetricsViewAggregationMeasure {
     MetricsViewAggregationMeasureComputeComparisonDelta comparison_delta = 8;
     MetricsViewAggregationMeasureComputeComparisonRatio comparison_ratio = 9;
     MetricsViewAggregationMeasureComputePercentOfTotal percent_of_total = 10;
+    MetricsViewAggregationMeasureURI uri = 11;
```
Comment (begelundmuller) [inline proto/rill/runtime/v1/queries.proto:407]: The type name should be `MetricsViewAggregationMeasureComputeURI` for consistency with the other "compute" options.

File: runtime/metricsview/query.go [hunk 2]
```diff
@@ -32,13 +32,18 @@ type Dimension struct {
...
 }
 
 type DimensionComputeTimeFloor struct {
 	Dimension string    `mapstructure:"dimension"`
 	Grain     TimeGrain `mapstructure:"grain"`
 }
 
+type DimensionComputeURI struct {
+	Dimension string `mapstructure:"dimension"`
+}
```
Comment (begelundmuller) [inline runtime/metricsview/query.go:45]: Seems like this isn't used?

**Review by begelundmuller — CHANGES_REQUESTED** [review]

**Review by begelundmuller — APPROVED** [review]

**Pull request merged**